### PR TITLE
Fix security vulnerability by getting rid of text/template 

### PR DESCRIPTION
**Pull request by kislaykishore**
Move to use safetext/yamltemplate

### Description
This change fixes a security vulnerability associated with text/template by replacing it with safetext/yamltemplate.

### Link to the issue in case of a bug fix.
b/344125887

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -62,6 +62,7 @@ require (
 	github.com/golang/protobuf v1.5.4 // indirect
 	github.com/google/renameio/v2 v2.0.0 // indirect
 	github.com/google/s2a-go v0.1.7 // indirect
+	github.com/google/safetext v0.0.0-20240104143208-7a7d9b3d812f // indirect
```
Comment (raj-prince) [inline go.mod:65]: Why is it in indirect? Did you run `go mod tidy`?

**Review by raj-prince — COMMENTED** [review]
Generally, we run presubmit_perf test when do any change in go.dependency.

**Comment by kislaykishore** [comment]
Performance benchmarks:
image

**Comment by kislaykishore** [comment]
> Generally, we run presubmit_perf test when do any change in go.dependency.

Done.

**Review by raj-prince — APPROVED** [review]

**Pull request merged**